docs/rfc/block-registration.md

```diff
@@ -0,0 +1,443 @@
+
+The following properties are going to be supported for backward compatibility reasons on the client-side only. Some of them might be replaced with alternative APIs in the future:
+ - `edit` - see the [Edit and Save](/docs/designers-developers/developers/block-api/block-edit-save.md) documentation for more details.
+ - `save` - see the [Edit and Save](/docs/designers-developers/developers/block-api/block-edit-save.md) documentation for more details.
+ - `transforms` - see the [Transforms](/docs/designers-developers/developers/block-api/block-registration.md#transforms-optional) documentation for more details.
+ - `deprecated` - see the [Deprecated Blocks](/docs/designers-developers/developers/block-api/block-deprecation.md) documentation for more details.
+ - `supports` - see the [block supports](/docs/designers-developers/developers/block-api/block-registration.md#supports-optional) documentation page for more details.
+ - `merge` - undocumented as of today. Its role is to handle merging multiple blocks into one.
+ - `getEditWrapperProps` - undocumented as well. Its role is to inject additional props to the block edit's component wrapper.
+ 
+**Example**:
+```js
+wp.blocks.registerBlockType( 'my-block/name', {
+	edit: function() {
+		// Edit definition goes here.
+	},
+	save: function() {
+		// Save definition goes here.
+	},
+	supports: {
+		html: false
+	}
+} );
+``` 
+
+In the case of [dynamic blocks](/docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md) supported by WordPress, it should be still possible to register `render_callback` property using [`register_block_type`](https://developer.wordpress.org/reference/functions/register_block_type/) function on the server.
+
+## Assets
+
+### `WPDefinedAsset`
+
+The `WPDefinedAsset` type is a subtype of string, where the value must represent an absolute or relative path to a JavaScript or CSS file.
+
+**Example:**
+
+In `block.json`:
+```json
+{ "editorScript": "build/editor.js" }
+```
+
+#### WordPress context
+
+In the context of WordPress, when a block is registered with PHP, it will automatically register all scripts and styles that are found in the `block.json` file.
+
+That's why, the `WPDefinedAsset` type has to offer a way to mirror also the shape of params necessary to register scripts and styles using [`wp_register_script`](https://developer.wordpress.org/reference/functions/wp_register_script/) and [`wp_register_style`](https://developer.wordpress.org/reference/functions/wp_register_style/), and then assign these as handles associated with your block using the `script`, `style`, `editor_script`, and `editor_style` block type registration settings.
+
+It's possible to provide an object which takes the following shape:
+- `handle` (`string`) - the name of the script. If omitted, it will be auto-generated.
+- `dependencies` (`string[]`) - an array of registered script handles this script depends on.  Default value: `array()`.
```

Comment:
It's a bit confusing whether we're talking about the default value in the context of `wp_register_block` or as in the `asset.json` file, but if it's considered more as the latter, it may be more appropriate to write this as:

```suggestion
- `dependencies` (`string[]`) - an array of registered script handles this script depends on.  Default value: `[]`.
```

(Which also shares meaning in PHP, assuming PHP 5.4+)

Reply:
I copied it from WordPress docs, it should be updated to align with JSON format :)